requests/adapters.py

```
@@ -328,7 +331,21 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
 
         chunked = not (request.body is None or 'Content-Length' in request.headers)
 
-        timeout = TimeoutSauce(connect=timeout, read=timeout)
+        if isinstance(timeout, tuple):
+            try:
+                connect, read = timeout
+                timeout = TimeoutSauce(connect=connect, read=read)
+            except ValueError as e:
+                try:
+                    err = ("Invalid timeout {0}. Pass a (connect, read) "
+                           "timeout tuple, or a single float to set "
+                           "both timeouts to the same value".format(timeout))
+                except Exception:
+                    # string formatting failed.
+                    raise ValueError(e)
```

Ah, who cares, if the user passed us something so insane that we can't format the string then let's just throw that exception.